### ISSUE TITLE
[FEATURE] Submit expense through Open Collective.

### ISSUE THREAD

**Issue by aerugo** [cobudget/cobudget#961]
When the Open Collective integration is active for a round ( see #959 ) we want to let the user submit expenses directly to Open Collective. Expense submissions are being worked on in #960 and will be possible without the integration, but will be much more powerful when combined with Open Collective.

Since we want Cobudget to keep track of which expenses to the collective belong to which approved bucket on Cobudget, we need to pass the expense ID back to Cobudget once it has been created. This rules out just linking to the expense submission page on Open Collective.

As far as I understand we have three options:

1) Replicate the entire expense submission UX in Cobudget and call the API.
2) Make the expense submission flow on Open Collective accessible to third party sites, returning the expense ID after submission.
3) Build a functionally reduced expense submission flow that is simpler than the experience on Open Collective, with enough information to still be able to submit an expense to Open Collective from Cobudget.
4) Store a hidden "tag" on the expense in Open Collective. This could work by Cobudget simply linking to the expense submission page, but passing the bucket ID on Cobudget as a parameter. If Open Collective could store that parameter with the expense as sort of hidden tag, and if Cobudget could query expenses with that hidden tag, then that could probably solve the problem.

Option 1 is complicated, though we could probably do it with the budget available since we have [access to the expense components](https://github.com/opencollective/opencollective-frontend/tree/main/components/expenses). However, this is not ideal as it will eventually break or get outdated.

Option 3 will leave the users of Cobudget frustrated, and is generally a poor solution.

Option 4 is clunky and not very elegant, but it would probably work and seems like the simplest option.

Option 2 has the major drawback of requiring support from the Open Collective engineering team. On the other hand, it could prove useful to other third party sites as well.

@alinauroz @gustavlrsn - what do you think we should do?

**Comment by Betree** [comment]
> Store a hidden "tag" on the expense in Open Collective. This could work by Cobudget simply linking to the expense submission page, but passing the bucket ID on Cobudget as a parameter. If Open Collective could store that parameter with the expense as sort of hidden tag, and if Cobudget could query expenses with that hidden tag, then that could probably solve the problem.

You could use `privateMessage` to store that, but it's currently impossible to query expenses based on this field. I think we could allow search in this field if the requester has admin permissions on the account.

> Make the expense submission flow on Open Collective accessible to third party sites, returning the expense ID after submission.

Based on our experience doing this for the [contribution flow](https://docs.opencollective.com/help/collectives/embed-contribution-flow), it's usually difficult to integrate this kind of solution with an existing design. The expense page is even more complex; I'm personally not confident with this approach, but not opposed to it if it gets prioritized as a real project.

**Comment by gustavlrsn** [comment]
Yeah, I think it would be best to avoid any reimplementation of the expense flow UI. Linking to Open Collective seems like the correct approach.

Using a tag or the `privateMessage` to achieve that it could make sense as a first step, but would love something more robust and easier to use across the board.

> > Make the expense submission flow on Open Collective accessible to third party sites, returning the expense ID after submission.
> 
> Based on our experience doing this for the [contribution flow](https://docs.opencollective.com/help/collectives/embed-contribution-flow), it's usually difficult to integrate this kind of solution with an existing design. The expense page is even more complex; I'm personally not confident with this approach, but not opposed to it if it gets prioritized as a real project.

I would love to see something like Stripe Checkout for Open Collective, i.e. not an embed, since those are often harder to work with as you say, but rather a scaled down separate page with just the expense (or contribute) flow, which then redirects you back to where you came from (with perhaps a `successRedirectUrl` and `cancelRedirectUrl` options).

I guess this is mostly what is already there for the contribution flow, except the redirects. 

Also, having the ability to store metadata on expenses and orders, to save the bucket ID would be nice. This data could then be included in the [Open Collective webhooks](https://docs.opencollective.com/help/collectives/collective-settings/integrations#webhooks-generic-slack-discord), that Cobudget can listen for to save the expense ID to the bucket.

**Comment by Betree** [comment]
Added some specs to enable the solution (3) in https://github.com/opencollective/opencollective/issues/6451
